### PR TITLE
fix: Cloud Ops permission errors

### PR DESCRIPTION
**Pull request by daniel-sanche**
Over the last few weeks, we've been seeing missing data in the Cloud Ops products on new sandboxes. I thought this was due to our outdated version of Istio, but after some experimentation, it looks like it was due to incorrect permissions on the GKE cluster service account

This PR fixes the issue by adding extra roles to the default compute instance service account, ensuring GCP observability client libraries can send data to Cloud Ops

### CODE REVIEW

File: terraform/03_gke_cluster.tf [hunk 1]
```diff
@@ -142,6 +142,43 @@ data "google_compute_default_service_account" "default" {
...
+resource "google_project_iam_member" "trace_role" {
+  project = data.google_project.project.project_id
+  role    = "roles/cloudtrace.admin"
+  member  = "serviceAccount:${data.google_compute_default_service_account.default.email}"
+  depends_on = [data.google_compute_default_service_account.default]
+}
+
+resource "google_project_iam_member" "monitoring_role" {
+  project = data.google_project.project.project_id
+  role    = "roles/monitoring.editor"
```
Comment (arbrown) [inline terraform/03_gke_cluster.tf:155]: If you want to limit permissions as much as possible, could this be `roles/monitoring.metricWriter` or does it need read access for something?
https://cloud.google.com/monitoring/access-control

File: terraform/03_gke_cluster.tf [hunk 2]
```diff
@@ -142,6 +142,43 @@ data "google_compute_default_service_account" "default" {
   ]
 }
 
+# Give service account Observability permissions
+resource "google_project_iam_member" "trace_role" {
+  project = data.google_project.project.project_id
+  role    = "roles/cloudtrace.admin"
```
Comment (arbrown) [inline terraform/03_gke_cluster.tf:148]: Could possibly be `roles/cloudtrace.agent`

**Review by arbrown — APPROVED** [review]
Consider tightening the permissions if possible.  I _think_ that the permissions needed for the service account could be somewhat reduced on the first two entries, but I'm not as close to the project, so I could be missing something and I'll trust your judgment.

File: terraform/03_gke_cluster.tf (hunk 2 above)
Comment (arbrown) [reply terraform/03_gke_cluster.tf:148]: Please excuse the double review.  Somehow this comment got left out of my first.

File: terraform/03_gke_cluster.tf (hunk 1 above)
Comment (daniel-sanche) [reply terraform/03_gke_cluster.tf:155]: Good catch, I meant to come back to these to tighten the permissions, but must have been distracted by a different PR

**Pull request merged**